a10mini-uboot.dts: give the A10 panel its ID (00 80 00) and add the V4 panel (ID 55 21 07, from
a10miniv4-uboot.dts) as a disabled sibling, for the panel-ID check (rk3326/uboot/panel-id).

--- a/arch/arm/dts/a10mini-uboot.dts	2026-10-22 16:24:24.629584145 -0700
+++ b/arch/arm/dts/a10mini-uboot.dts	2026-10-22 16:49:42.662039842 -0700
@@ -174,6 +174,9 @@
 	panel@0 {
 		compatible = "elida,kd35t133", "simple-panel-dsi";
 		reg = <0>;
+		/* Panel-ID check: this panel answers DCS 0x04 with 00 80 00 */
+		id-reg = <0x04>;
+		id = [00 80 00];
 
 		/* pwm1 backlight */
 		backlight = <&backlight>;
@@ -185,8 +188,9 @@
 		/* Rev 0.2 LCD reset gpio GPIO3.C0 */
 		reset-gpios = <&gpio3 RK_PC0 GPIO_ACTIVE_LOW>;
 
-		prepare-delay-ms = <0>;
-		reset-delay-ms = <0>;
+		/* The V4 panel's delays, so its ID read can answer too */
+		prepare-delay-ms = <10>;
+		reset-delay-ms = <10>;
 		init-delay-ms = <80>;
 		enable-delay-ms = <0>;
 		disable-delay-ms = <10>;
@@ -244,6 +248,84 @@
 			};
 		};
 	};
+
+	/*
+	 * The A10 Mini V4 panel (720x540, ID 55 21 07), from a10miniv4-uboot.dts; only
+	 * the panel-ID check reads this node.
+	 */
+	panel@1 {
+		compatible = "elida,kd35t133", "simple-panel-dsi";
+		reg = <0>;
+		status = "disabled";
+		id-reg = <0x04>;
+		id = [55 21 07];
+
+		/* pwm1 backlight */
+		backlight = <&backlight>;
+		/* RK817 LDO7 */
+		backlight-supply = <&vcc_backlight>;
+		/* RK817 LDO8 */
+		power-supply = <&vcc_lcd>;
+
+		/* Rev 0.2 LCD reset gpio GPIO3.C0 */
+		reset-gpios = <&gpio3 RK_PC0 GPIO_ACTIVE_LOW>;
+
+		prepare-delay-ms = <10>;
+		reset-delay-ms = <10>;
+		init-delay-ms = <80>;
+		enable-delay-ms = <26>;
+		disable-delay-ms = <10>;
+		unprepare-delay-ms = <60>;
+
+		/* LCD size */
+		width-mm = <43>;
+		height-mm = <57>;
+
+		dsi,flags = <(MIPI_DSI_MODE_VIDEO           |
+			      MIPI_DSI_MODE_VIDEO_BURST     |
+			      MIPI_DSI_MODE_LPM             |
+			      MIPI_DSI_MODE_EOT_PACKET)>;
+
+		dsi,format = <MIPI_DSI_FMT_RGB888>;
+		dsi,lanes = <2>;
+		/*
+			- panel-init-sequence data description -
+			  mipi_cmd, delay(ms), len, lcd_cmd, lcd_data...
+
+			- mipi_cmd description -
+			  05 : MIPI_DSI_SHORT_WRITE
+			  15 : MIPI_DSI_SHORT_WRITE_PARAM
+			  39 : MIPI_DSI_LONG_WRITE		
+		*/
+		panel-init-sequence = [39 00 05 ff aa 55 a5 80 39 00 06 f0 55 aa 52 08 00 15 00 02 36 00 39 00 03 b1 68 27 15 00 02 b5 0f 15 00 02 b6 0c 39 00 05 b8 01 02 08 02 15 00 02 ba 02 39 00 03 bb 24 24 39 00 03 bc 00 00 39 00 06 bd 02 68 20 20 00 39 00 06 be 02 68 20 20 00 39 00 07 cc 41 36 87 00 00 00 39 00 07 ce 10 10 10 10 10 10 39 00 06 f0 55 aa 52 08 06 39 00 03 b0 2e 2e 39 00 03 c2 17 11 05 78 01 11 05 78 01 29];
+
+		/* 28: DisplayOff */
+		/* 10: SleepIn */
+		panel-exit-sequence = [
+			05 14 01 28
+			05 0A 01 10
+		];
+
+		display-timings {
+			native-mode = <&v4_timing>;
+
+			v4_timing: timing0 {
+				clock-frequency = <30000000>;
+				hactive = <720>;
+				vactive = <540>;
+				hfront-porch = <32>;
+				hsync-len = <40>;
+				hback-porch = <32>;
+				vfront-porch = <10>;
+				vsync-len = <20>;
+				vback-porch = <38>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <0>;
+				pixelclk-active = <0>;
+			};
+		};
+	};
 };
 &dsi_in_vopb {
 	status = "okay";
